eodag/api/constraints.py

```diff
@@ -0,0 +1,165 @@
+import logging
+from copy import deepcopy
+from typing import Any, Dict, Union
+
+import requests
+
+from eodag.plugins.apis.base import Api
+from eodag.plugins.search.base import Search
+from eodag.utils import USER_AGENT
+from eodag.utils.exceptions import NotAvailableError
+
+logger = logging.getLogger("eodag.constraints")
+
+
+def get_constraint_queryables_with_additional_params(
+    constraints: list,
```

Comment:
Type hinting of the method is incomplete:
- list should be qualified type: List[something. Same for Dict
- constraint type is unknown
- queryables type is Dict[unknown, unknown]